scripts/zones/Oldton_Movalpolos/npcs/Tarnotik.lua

```diff
@@ -11,7 +11,11 @@ local entity = {}
 
 entity.onTrade = function(player, npc, trade)
     if player:getCurrentMission(xi.mission.log_id.COP) >= xi.mission.id.cop.THREE_PATHS and npcUtil.tradeHas(trade, 1725) then
-        player:startEvent(32)
+        if player:hasKeyItem(xi.ki.SHAFT_2716_OPERATING_LEVER) then
+            player:startEvent(54)
+        else
+            player:startEvent(32)
+        end
```

Comment:
Think we discovered that this should be in on Trigger than on the trade. Just not sure how he should react if you trade a snow lily and have the KI since he shouldn't teleport you.

Reply:
I'll get a good capture for it soon. I'm near this mission on retail and I can figure out the proper mechanics. I'll bet he behaves the same way twinkbrix does. Won't teleport you until you forfeit the lever!

Reply:
I think so too. Just not sure how the NPC should react or what their dialogue should be when you try to.

Reply:
Got the capture and pushed the appropriate behavior!